modules/partition-replicator/src/integrationTest/java/org/apache/ignite/internal/partition/replicator/ItBuildIndexTest.java

```diff
@@ -0,0 +1,137 @@
+import org.apache.ignite.internal.hlc.HybridClock;
+import org.apache.ignite.internal.index.message.IndexMessageGroup;
+import org.apache.ignite.internal.index.message.IndexMessagesFactory;
+import org.apache.ignite.internal.index.message.IsNodeFinishedRwTransactionsStartedBeforeRequest;
+import org.apache.ignite.internal.partition.replicator.fixtures.Node;
+import org.apache.ignite.internal.table.TableTestUtils;
+import org.apache.ignite.internal.table.TableViewInternal;
+import org.apache.ignite.table.KeyValueView;
+import org.apache.ignite.table.Tuple;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+// TODO: https://issues.apache.org/jira/browse/IGNITE-22522 remove this test after the switching to zone-based replication
+/**
+ * Tests building indices for colocation track.
+ */
+@Timeout(60)
+public class ItBuildIndexTest extends ItAbstractColocationTest {
+    private static final IndexMessagesFactory FACTORY = new IndexMessagesFactory();
+
+    @Test
+    public void testBuildIndex() throws Exception {
+        // Prepare a single node cluster.
+        startCluster(1);
+        Node node = getNode(0);
+
+        String zoneName = "test-zone";
+        createZone(node, zoneName, 1, 1);
+
+        String tableName = "TEST_TABLE";
+        createCustomTable(node, zoneName, tableName);
+        int tableId = TableTestUtils.getTableId(node.catalogManager, tableName, node.hybridClock.nowLong());
+
+        // Test node does not create IndexNodeFinishedRwTransactionsChecker, so the following code is needed to unblock index building.
+        // It's easier than creating a real service with all its dependencies.
+        node.clusterService.messagingService().addMessageHandler(
+                IndexMessageGroup.class,
+                (message, sender, correlationId) -> {
+                    if (message instanceof IsNodeFinishedRwTransactionsStartedBeforeRequest) {
+                        node.clusterService.messagingService().respond(
+                                sender,
+                                FACTORY.isNodeFinishedRwTransactionsStartedBeforeResponse().finished(true).build(),
+                                correlationId
+                        );
+                    }
+                });
+
+        TableViewInternal tableViewInternal = node.tableManager.table(tableId);
+        KeyValueView<Tuple, Tuple> tableView = tableViewInternal.keyValueView();
+
+        node.transactions().runInTransaction(tx -> {
+            Tuple key = Tuple.create().set("KEY", 1L);
+            Tuple value = Tuple.create().set("VAL", 1).set("DOUBLEVAL", 1.0);
+            tableView.putAll(tx, Map.of(key, value));
+        });
+
+        // This async transaction is needed to update safe time, just because the idle safe time propagation is not implemented yet.
+        // https://issues.apache.org/jira/browse/IGNITE-22620
+        CompletableFuture.runAsync(() -> {
```

Comment:
Why don't you do anything with the return futures?

Reply:
It does not matter from the test point of view. We just need to update `safe-time` as described in the comment until [IGNITE-22620](https://issues.apache.org/jira/browse/IGNITE-22620)